source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -592,6 +624,63 @@ FilterDataStatus Filter::onData(ProcessorState& state, Buffer::Instance& data, b
   return result;
 }
 
+FilterHeadersStatus
+Filter::sendHeadersInObservabilityMode(Http::RequestOrResponseHeaderMap& headers,
+                                       ProcessorState& state, bool end_stream) {
+  switch (openStream()) {
+  case StreamOpenState::Error:
+    return FilterHeadersStatus::StopIteration;
+  case StreamOpenState::IgnoreError:
+    return FilterHeadersStatus::Continue;
+  case StreamOpenState::Ok:
+    // Fall through
+    break;
+  }
+
+  ProcessingRequest req;
+  req.set_observability_mode(true);
+  addAttributes(state, req);
```

Comment:
Should there be some refactoring to have a common place the headers request is assembled? The only difference should be the setting of observability_mode to true right?

Reply:
Done.